client_test.go

```diff
@@ -158,10 +159,80 @@ func runClient(ctx context.Context, t *testing.T, client *Client[pgx.Tx]) {
 func runNewTestClient(ctx context.Context, t *testing.T, config *Config) *Client[pgx.Tx] {
 	t.Helper()
 	client := newTestClient(ctx, t, config)
-	runClient(ctx, t, client)
+	startClient(ctx, t, client)
 	return client
 }
 
+func Test_Client(t *testing.T) {
```

Comment:
Started a new branch of client tests because it's not clear what should go where in here anymore, and I'm trying to put in something where new tests can be added without starting a fresh test every time. We'll try to take advantage of `JobArgsReflectKind` and `WorkFunc` in subtests so that args and work definition can be colocated right inside the test rather than being floating around in global state somewhere.

`StartInsertAndWork` below is a basic client test (not related to the `Schema` change) added to show roughly what this look like.